src/frontend/js/components/PaymentButton/index.spec.tsx

```diff
@@ -62,6 +62,14 @@ jest.mock('utils/context', () => ({
 
 jest.mock('./components/PaymentInterfaces');
 
+type WrapperProps = PropsWithChildren<{
+  // hallucination:
+  // eslint-disable-next-line react/no-unused-prop-types
+  client?: QueryClient;
+  product: CredentialProduct | CertificateProduct;
+  orderGroup?: OrderGroup;
+}>;
+
```

Comment:
This is not an hallucination.
The `SaleTunnelWrapper` never uses the `client` prop. In order to fix that you should create two types.

```ts
type SaleTunnelWrapperProps = PropsWithChildren<{
  product: CredentialProduct | CertificateProduct;
  orderGroup?: OrderGroup;
}>;
type WrapperProps = PropsWithChildren<
  SaleTunnelWrapperProps & {
    client?: QueryClient;
  }
>;
```

Reply:
I used `Exclude`